marklogic/src/main/scala/quasar/physical/marklogic/fs/data.scala

```diff
@@ -48,19 +49,107 @@ object data {
       }
   }
 
+  def toXml[F[_]: MonadErrMsg](data: Data): F[Elem] = {
+    def typeAttr(tpe: String): Attribute =
+      Attribute(ejsBinding.prefix, ejsonType.local.shows, tpe, Null)
+
+    def elem(name: QName, attrs: MetaData, children: Node*): Elem =
+      Elem(name.prefix.map(_.shows).orNull, name.local.shows, attrs, TopScope, true, children: _*)
+
+    def typed(name: QName, tpe: String, children: Node*): F[Elem] =
+      elem(name, typeAttr(tpe), children: _*).point[F]
+
+    def rootElem(name: QName, attrs: MetaData, children: Node*): Elem =
+      Elem(name.prefix.map(_.shows).orNull, name.local.shows, attrs, ejsBinding, true, children: _*)
+
+    def literal(tpe: String, children: Node*): F[Elem] =
+      rootElem(ejsonLiteral, typeAttr(tpe), children: _*).point[F]
+
+    def toXml0(elementName: QName): Data => F[Elem] = {
+      case Data.Binary(bytes) => ???
+      case Data.Bool(b)       => typed(elementName, "boolean", Text(b.fold("true", "false")))
+      case Data.Date(d)       => ???
+      case Data.Dec(d)        => typed(elementName, "decimal", Text(d.toString))
+      case Data.Id(id)        => typed(elementName, "id", Text(id))
+      case Data.Int(i)        => typed(elementName, "integer", Text(i.toString))
+      case Data.Interval(ivl) => ???
+      case Data.Null          => typed(elementName, "null")
+      case Data.Str(s)        => typed(elementName, "string", Text(s))
```

Comment:
I don't think we need to do anything special for String, any reason this is `typed`?